prediction/independent_poisson_model.py

```diff
@@ -14,42 +15,129 @@ class IndependentPoissonModel(object):
     """Models goals to be scored as two independent Poisson R.Vs. """
     def predict(self, home_team, away_team):
 
+        l_A,l_B = self.getExpectation(home_team,away_team)
+
+        score_dict = self.run_simulations(l_A, l_B, self.NUM_ITERS)
+
+        return score_dict
+
+
+    def getExpectation(self,home_team,away_team):
         home_team = home_team.lower()
         away_team = away_team.lower()
 
         home_team_elo = get_current_elo(home_team)
         away_team_elo = get_current_elo(away_team)
+        home_team_matches = filter_matches(home_team)
+        away_team_matches = filter_matches(away_team)
```

Comment:
This is an improvement (because within the function, you only hit the data base twice instaed of 4 times) but I doubt this is a significant gain.

What you need to fix is the redundant DB access between each tournament simulation (i.e. the process of running all games to produce a World Cup winner). Because, if you think about it, once you compute uA_B, uB_A, vA_B, vB_A once for some country, you don't need to compute it again between different tournament simulations.